rfcs/033-lineage-graph.md

```diff
@@ -0,0 +1,101 @@
+v0 of lineage is already available in the Amundsen project today in the form of a list.
+
+## Guide-level Explanation (aka Product Details)
+
+This feature will expose upstream and downstream tables and columns within the Table Details page, using a navigable graph visualization.
+
+- Lineage: Lineage is a term that describes the flow of data from one entity to another. While this term can broadly include everything from services, events, ETLs, and dashboards, we will focus on table-to-table and column-to-column data lineage in this RFC.
+- Upstream: Upstream is a relative term that describes data sources from which we inherit. Data flows from upstream to downstream.
+- Downstream: Downstream is a relative term that describes data entities that consume our data.
+- Level: The depth of the graph in terms of relationships with another table. By default, this will only display first-level (immediate) upstream and downstream tables.
+- Direction: Upstream, Downstream, or Both. Users will be able to filter out the graph based on the direction. by default, we will display `Both` upstream and downstream tables.
+
+Since this is a UI/UX feature, more details about the implementation in the next section.
+
+## UI/UX-level Explanation
+
+> Explain the UI changes that your proposal would need (if applicable). This could mean:
+
+The feature will be disabled by default and can be enabled via a configuration flag. (This is how Amundsen enables/disables lineage today)
+Once enabled,
+
+- a new CTA/Button will appear on the top right of the table detail page.
+- a new icon in front of the column name (to support column-level lineage).
+
+Clicking on any of the above will navigate the user to a new page displaying lineage graph along with the filters to select the `level` and `direction` for the graph.
+
+The sidebar will have the filters available and will be used for other interactions like displaying a complete list of nodes, and/or view the metadata of a selected node.
+
+Interaction Details:
+
+- By default, we will display 1 level (or depth) of the lineage. Users will be able to control the level by clicking Expand/Collapse via each node. Or simply use the filter to change the number of levels shown.
+- User will be able to Pane and Zoom, and also reset the whole graph.
+- Clicking any node will display the metadata details of that particular node in the sidebar.
+
+Important to note: We will not display the complete list of upstream/downstream tables. We will only display X number of tables by default based on their usage and will control the number by a config flag.
+
+See the Screenshots in [this folder](https://github.com/amundsen-io/rfcs/tree/98c240d29d8644a81b4eaf6a1481a126d08b5732/assets/033).
+
+## Reference-level Explanation (aka Technical Details)
+
+This does not change anything on the backend and uses the existing endpoints exposed in the v0 of lineage.
```

Comment:
I believe some new filtering APIs will be required, as will limiting of responses based on usage?